src/rpc/methods/eth.rs

```diff
@@ -2847,6 +2894,117 @@ impl RpcMethod<1> for EthGetFilterLogs {
     }
 }
 
+pub enum EthGetFilterChanges {}
+impl RpcMethod<1> for EthGetFilterChanges {
+    const NAME: &'static str = "Filecoin.EthGetFilterChanges";
+    const NAME_ALIAS: Option<&'static str> = Some("eth_getFilterChanges");
+    const N_REQUIRED_PARAMS: usize = 1;
+    const PARAM_NAMES: [&'static str; 1] = ["filter_id"];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Write;
```

Comment:
Why does it need `write` permsissions?

Reply:
Write permission is required because we update the `collected` field for filter on every filter execution to return event logs since last poll

Reply:
Lotus uses `perm:read`. I don't think this method should update the filter store.

Reply:
https://github.com/filecoin-project/lotus/blob/a474d459b2b3be1d888f710e455c843b14e580dc/documentation/en/api-v1-unstable-methods.md?plain=1#L1723-L1728

Reply:
I think this should be `perm:write` since we need to return event logs since last filter execution, we need to update `collected` field for filter which is executed

Reply:
Ah, I see! In that case, Lotus isn't setting the permission properly, and that's an issue.